### PR TITLE
UHF-9113: sync changes from main to v4 branch

### PR DESCRIPTION
**Pull request by hyrsky**
# [UHF-9113](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9113)

## What was done

We need to release a new major revision for Drupal 10, since most of the patches in this repo are not compatible. My proposal is to have the drupal 10 related changes in a separate branch until we have upgraded everything. This PR adds automation to keep the `4.x` and `main` branches in sync with automatic pull requests, so we don't have to do manual porting.

## How to install
* N/A

## How to test

* [ ] Check that code follows our standards

[UHF-9113]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/v4-sync.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create pull request
+        run: |
+          gh pr create \
+            --base main \
+            --head 1.x \
```
Comment (tuutti) [inline .github/workflows/v4-sync.yml:18]: Should this be 4.x?

**Review by tuutti — CHANGES_REQUESTED** [review]

File: .github/workflows/v4-sync.yml (hunk 1 above)
Comment (hyrsky) [reply .github/workflows/v4-sync.yml:18]: Oops :D

Comment (hyrsky) [reply .github/workflows/v4-sync.yml]: I changed this to rebase the changes to 4.x branch instead of creating a pull request.

**Review by tuutti — APPROVED** [review]
Looks good to me

**Pull request merged**